commons-core/src/main/scala/com/avsystem/commons/serialization/cbor/CborAdtMetadata.scala

```diff
@@ -253,3 +253,21 @@ trait CborAdtInstances[T] {
   def cborCodec: GenObjectCodec[T] =
     metadata.setup(_.validate()).adjustCodec(stdCodec)
 }
+
+trait CborAdtPolyInstances[C[_]] {
+  def stdCodec[T: GenCodec]: GenObjectCodec[C[T]]
+  def metadata[T: GenCodec]: CborAdtMetadata[C[T]]
+  def cborCodec[T: GenCodec](validate: Boolean): GenObjectCodec[C[T]] =
```

Comment:
What's the point of the `validate` param now?

Reply:
I thought about some hypothetical manual codec creation where a user may want to force validation.

Reply:
Why? This trait's sole purpose is to be used by `HasPolyCborCodec`

Reply:
You can just remove `def cborCodec` from this trait entirely and do all the work in `HasPolyCborCodec`